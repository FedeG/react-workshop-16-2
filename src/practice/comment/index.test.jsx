import React from 'react';
import MockDate from 'mockdate';
import renderer from 'react-test-renderer';

import Comment from './Comment';

MockDate.set('3/5/1995');

describe('Comment', () => {
  test('initial state', () => {
    const author = {
      name: 'Test',
      avatarUrl: 'avatarTestUrl',
    };

    const component = renderer.create(
      <Comment
        author={author}
        date={new Date()}
        comments={['buenos dias', 'buenas tarde', 'buenas noches']}
      />,
    );

    const instance = component.getInstance();
    expect(instance.state.date).toEqual(new Date());
    expect(instance.state.showDate).toBeFalsy();

    /*
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    */
  });
});
