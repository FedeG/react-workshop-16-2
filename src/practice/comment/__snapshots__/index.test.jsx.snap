// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment initial state 1`] = `
<div
  className="Comment"
>
  <div
    className="UserInfo"
  >
    <div
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <img
        alt="Test"
        className="Avatar"
        src="avatarTestUrl"
      />
    </div>
    <div
      className="UserInfo-name"
    >
      Test
    </div>
  </div>
  <div
    className="Comment-text"
  >
    <div>
      <h1>
        Comentarios:
      </h1>
      <div
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        <img
          alt="Test"
          className="Avatar"
          src="avatarTestUrl"
        />
      </div>
      <h4>
        buenos dias
      </h4>
      <h4>
        buenas tarde
      </h4>
      <h4>
        buenas noches
      </h4>
    </div>
  </div>
  <div
    className="Comment-date"
  >
    Sun Mar 05 1995
  </div>
</div>
`;

exports[`Comment when click, date should change 1`] = `
<div
  className="Comment"
>
  <div
    className="UserInfo"
  >
    <div
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <img
        alt="Test"
        className="Avatar"
        src="avatarTestUrl"
      />
    </div>
    <div
      className="UserInfo-name"
    >
      Test
    </div>
  </div>
  <div
    className="Comment-text"
  >
    <div>
      <h1>
        Comentarios:
      </h1>
      <div
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        <img
          alt="Test"
          className="Avatar"
          src="avatarTestUrl"
        />
      </div>
      <h4>
        buenos dias
      </h4>
      <h4>
        buenas tarde
      </h4>
      <h4>
        buenas noches
      </h4>
    </div>
  </div>
  <div
    className="Comment-date"
  >
    Sun Mar 05 1995
  </div>
</div>
`;

exports[`Comment when click, date should change 2`] = `
<div
  className="Comment"
>
  <div
    className="UserInfo"
  >
    <div
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <img
        alt="Test"
        className="Avatar"
        src="avatarTestUrl"
      />
    </div>
    <div
      className="UserInfo-name"
    >
      Test
    </div>
  </div>
  <div
    className="Comment-text"
  >
    <div>
      <h1>
        Comentarios:
      </h1>
      <div
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        <img
          alt="Test"
          className="Avatar"
          src="avatarTestUrl"
        />
      </div>
      <h4>
        buenos dias
      </h4>
      <h4>
        buenas tarde
      </h4>
      <h4>
        buenas noches
      </h4>
    </div>
  </div>
  <div
    className="Comment-date"
  >
    Sun Mar 05 1995
  </div>
  <div>
    <h1>
      Hello, world!
    </h1>
    <h2>
      Fecha: Tue Mar 05 1996
    </h2>
  </div>
</div>
`;

exports[`Comment when click, date should change 3`] = `
<div
  className="Comment"
>
  <div
    className="UserInfo"
  >
    <div
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <img
        alt="Test"
        className="Avatar"
        src="avatarTestUrl"
      />
    </div>
    <div
      className="UserInfo-name"
    >
      Test
    </div>
  </div>
  <div
    className="Comment-text"
  >
    <div>
      <h1>
        Comentarios:
      </h1>
      <div
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        <img
          alt="Test"
          className="Avatar"
          src="avatarTestUrl"
        />
      </div>
      <h4>
        buenos dias
      </h4>
      <h4>
        buenas tarde
      </h4>
      <h4>
        buenas noches
      </h4>
    </div>
  </div>
  <div
    className="Comment-date"
  >
    Sun Mar 05 1995
  </div>
  <div>
    <h1>
      Hello, world!
    </h1>
    <h2>
      Fecha: Wed Mar 05 1997
    </h2>
  </div>
</div>
`;

exports[`Comment when click, showDate should be true 1`] = `
<div
  className="Comment"
>
  <div
    className="UserInfo"
  >
    <div
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <img
        alt="Test"
        className="Avatar"
        src="avatarTestUrl"
      />
    </div>
    <div
      className="UserInfo-name"
    >
      Test
    </div>
  </div>
  <div
    className="Comment-text"
  >
    <div>
      <h1>
        Comentarios:
      </h1>
      <div
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        <img
          alt="Test"
          className="Avatar"
          src="avatarTestUrl"
        />
      </div>
      <h4>
        buenos dias
      </h4>
      <h4>
        buenas tarde
      </h4>
      <h4>
        buenas noches
      </h4>
    </div>
  </div>
  <div
    className="Comment-date"
  >
    Sun Mar 05 1995
  </div>
  <div>
    <h1>
      Hello, world!
    </h1>
    <h2>
      Fecha: Sun Mar 05 1995
    </h2>
  </div>
</div>
`;
